File: 0114-二叉树展开为链表/Solution.java

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

class Solution {
    public void flatten(TreeNode root) {
        if(root == null){ return; }
        if(root.left != null){ flatten(root.left); }
        if(root.right != null){ flatten(root.right); }
        if(root.left != null){
            TreeNode curr = root.left;
            while(curr.right != null){ curr = curr.right; }
            curr.right = root.right;
            root.right = root.left;
            root.left = null;
        }
    }
}

File: 0114-二叉树展开为链表/solution.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def flatten(self, root: TreeNode) -> None:
        """
        Do not return anything, modify root in-place instead.
        """
        if root is None:
            return None
        left = self.flatten(root.left)
        right = self.flatten(root.right)
        root.left = None
        root.right = left
        tail = root
        while tail.right:
            tail = tail.right
        tail.right = right
        return root